src/coreclr/src/vm/pefile.h

```diff
@@ -581,11 +581,13 @@ class PEFile
 
 #endif //!DACCESS_COMPILE
 
+    // Returns AssemblyLoadContext into which the current PEFile was loaded.
+    // It returns NULL for System.Private.CoreLib during early runtime initialization
+    // stage when the related AssemblyLoadContext does not exist yet.
     PTR_AssemblyLoadContext GetAssemblyLoadContext()
     {
         LIMITED_METHOD_CONTRACT;
 
-        _ASSERTE(m_pAssemblyLoadContext != NULL);
```

Comment:
I think this assert is good and it should stay.

Would it be possible to `AppDomain::GetCurrentDomain()->CreateBinderContext()` instead of `AppDomain::GetCurrentDomain()->GetTPABinderContext()` in `SetupAssemblyLoadContext`; and also delete the null checked for GetAssemblyLoadContext returning NULL from `AcquireCompositeImage` ?

Reply:
That could work, I'll give it a try.